consumer.go

```diff
@@ -119,9 +119,10 @@ func (c *consumer) ConsumePartition(topic string, partition int32, offset int64)
 		fetchSize: c.conf.Consumer.Fetch.Default,
 	}
 
-	if err := child.chooseStartingOffset(offset); err != nil {
-		return nil, err
+	if offset < -2 {
+		return nil, ConfigurationError("Invalid offset")
 	}
+	child.offset = offset
```

Comment:
This won't work, we need to call `GetOffset` to convert e.g. -1 to a real offset